recipes/libnghttp2/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_app": [True, False],
+               "with_hpack": [True, False],
+               "with_asio": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_app": True,
+                       "with_hpack": True,
+                       "with_asio": False}
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+        if self.options.with_asio and self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Build with asio and MSVC is not supported yet, see upstream bug #589")
+
+    def requirements(self):
+        self.requires.add("zlib/1.2.11")
+        if self.options.with_app:
+            self.requires.add("openssl/1.0.2t")
+            self.requires.add("c-ares/1.15.0")
+            self.requires.add("libev/4.25")
+            self.requires.add("libxml2/2.9.9")
+        if self.options.with_hpack:
+            self.requires.add("jansson/2.12")
+        if self.options.with_asio:
+            self.requires.add("boost/1.70.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "nghttp2-{0}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        cmake.definitions["ENABLE_SHARED_LIB"] = "ON" if self.options.shared else "OFF"
+        cmake.definitions["ENABLE_STATIC_LIB"] = "OFF" if self.options.shared else "ON"
+        cmake.definitions["ENABLE_HPACK_TOOLS"] = "ON" if self.options.with_hpack else "OFF"
+        cmake.definitions["ENABLE_APP"] = "ON" if self.options.with_app else "OFF"
+        cmake.definitions["ENABLE_EXAMPLES"] = "OFF"
+        cmake.definitions["ENABLE_PYTHON_BINDINGS"] = "OFF"
+        cmake.definitions["ENABLE_FAILMALLOC"] = "OFF"
+        # disable unneeded auto-picked dependencies
+        cmake.definitions["WITH_LIBXML2"] = "OFF"
+        cmake.definitions["WITH_JEMALLOC"] = "OFF"
+        cmake.definitions["WITH_SPDYLAY"] = "OFF"
+
+        cmake.definitions["ENABLE_ASIO_LIB"] = "ON" if self.options.with_asio else "OFF"
+
+        if self.options.with_app:
+            cmake.definitions['OPENSSL_ROOT_DIR'] = self.deps_cpp_info['openssl'].rootpath
+        if self.options.with_asio:
+            cmake.definitions['BOOST_ROOT'] = self.deps_cpp_info['boost'].rootpath
+        cmake.definitions['ZLIB_ROOT'] = self.deps_cpp_info['zlib'].rootpath
+
+        cmake.configure()
+        return cmake
+
+    def _build_with_autotools(self):
+        if self.options.with_app:
+            os.rename('c-ares.pc', 'libcares.pc')
```

Comment:
Where are these pkg-config coming from? Should this recipe us the `pkg_config` generator for this `with_app` option?

Reply:
It looks like that the name for the `c-ares` library should be `libcares` so Conan generates the proper `.pc` file. Or we should use the _components feature_ to give it a different name.

Reply:
we already have `cppinfo.name`, that should be enough, right?

Reply:
Anyway, this is not an issue of this recipe, but the `c-ares` one